### PR TITLE
ENCD-4905-add-search-view-tests

### PR DESCRIPTION
**Pull request by keenangraham**

### CODE REVIEW

File: src/encoded/tests/test_search_views.py [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+    assert 'columns' in r.json
+    assert 'clear_filters' in r.json
+    assert r.json['clear_filters'] == '/search/?type=Experiment'
+    assert r.json['@type'] == ['ExperimentCollection', 'Collection']
+    assert r.json['@context'] == '/terms/'
+
+
+def test_collection_listing_es_view_item(workbook, testapp):
+    r = testapp.get(
+        '/Experiment'
```
Comment (hitz) [inline src/encoded/tests/test_search_views.py]: this is an odd url.  Isn't the collection /experiments/?   /Experiment redirects.

File: src/encoded/tests/test_search_views.py [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+    assert len(r.json['@graph']) == 50
+    assert 'all' not in r.json
+
+
+def test_search_view_with_limit_zero(workbook, testapp):
+    r = testapp.get(
+        '/search/?type=Experiment&limit=0'
+    )
+    assert len(r.json['@graph']) == 0
+    assert 'all' in r.json
```
[the remark this answers is not in the document]
Comment (keenangraham) [reply src/encoded/tests/test_search_views.py]: A link to view all shows up if there are more results than you are viewing. If you are already viewing all then it shouldn't show up.

File: src/encoded/tests/test_search_views.py [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+    r = testapp.get('/searchv2_raw/?type=Experiment')
+    assert 'hits' in r.json
+    assert 'aggregations' in r.json
+    assert '_shards' in r.json
+    assert 'timed_out' in r.json
+    assert 'took' in r.json
+
+
+def test_search_raw_view_raw_response_limit_all(workbook, testapp):
+    r = testapp.get('/searchv2_raw/?type=*&limit=all')
```
[the remark this answers is not in the document]
Comment (keenangraham) [reply src/encoded/tests/test_search_views.py]: Yeah.

File: src/encoded/tests/test_search_views.py (hunk 1 above)
Comment (keenangraham) [reply src/encoded/tests/test_search_views.py]: This is testing the redirect.

**Comment by keenangraham** [comment]
Rebased over dev now that test improvements have been merged.

**Pull request merged**